src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,223 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class StringCalculator {
+
+    private static final String SPLIT_REGEX = " ";
+    private static final String MARK_ADD = "+";
+    private static final String MARK_SUBTRACTION = "-";
+    private static final String MARK_MULTIPLICATION = "*";
+    private static final String MARK_DIVISION = "/";
+    
+    private List<String> inputList;
+    private int baseValue = 0;
+    private int differentValue = 0;
+    private String mark;
+
+    public StringCalculator(String input) {
+        this.inputList = this.splitInput(input);
+    }
+
+    public List<String> getInputList() {
+        return inputList;
+    }
+
+    public void setInputList(List<String> inputList) {
+        this.inputList = inputList;
+    }
+
+    public int getBaseValue() {
+        return baseValue;
+    }
+
+    public void setBaseValue(int baseValue) {
+        this.baseValue = baseValue;
+    }
+
+    public int getDifferentValue() {
+        return differentValue;
+    }
+
+    public void setDifferentValue(int differentValue) {
+        this.differentValue = differentValue;
+    }
+
+    public String getMark() {
+        return mark;
+    }
+
+    public void setMark(String mark) {
+        this.mark = mark;
+    }
+
+    /**
+     * 입력 값을 쪼개어 리스트에 담습니다.
+     * @param input
+     * @return
+     */
+    public List<String> splitInput(String input) {
+        return Arrays.asList(input.split(StringCalculator.SPLIT_REGEX));
+    }
+
+    /**
+     * 입력 된 문자열 값을 계산합니다.
+     * @return
+     */
+    public int calculate() throws IllegalArgumentException {
+        for (int i = 0 ; i < this.getInputList().size() ; i++) {
+            this.saveValueAndMark(this.getInputList().get(i).trim(), i);
+            executeCalculate(i);
+        }
+        return this.getBaseValue();
+    }
+
+    /**
+     * 숫자 값 혹은 연산기호를 저장합니다.
+     * @param input
+     * @param index
+     */
+    public void saveValueAndMark(String input, int index) throws IllegalArgumentException {
+        System.out.println("set value : " + input);
+        this.isNotData(input);
+
+        if(this.isCalculateIndex(index)) {
+            this.setDifferentValue(Integer.parseInt(input));;
+            return;
+        }
+
+        if(this.isNumberDataIndex(index)) {
+            this.setBaseValue(Integer.parseInt(input));
+            return;
+        }
+
+        this.saveCalculateMark(input);
+    }
+
+    /**
+     * 데이터가 null이거나 빈 공백 문자일경우 오류를 발생시킵니다.
+     * @param input
+     * @throws IllegalArgumentException
+     */
+    public void isNotData(String input) throws IllegalArgumentException {
```

Comment:
```suggestion
    public void isNotData(String input) {
```
if 절에서 이미 throw 하고 있네요

Reply:
변경하였습니다 :)